shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -921,6 +927,42 @@ - (void)waitForFirstFrame:(NSTimeInterval)timeout
   });
 }
 
+- (FlutterEngine*)spawnWithEntrypoint:(NSString*)entrypoint {
+  assert(_shell);
+  FlutterEngine* result =
+      [[FlutterEngine alloc] initWithName:[_labelPrefix stringByAppendingString:@"-spawn"]
+                                  project:_dartProject.get()
+                   allowHeadlessExecution:_allowHeadlessExecution];
+
+  flutter::Settings settings = _shell->GetSettings();
+  if (entrypoint) {
```

Comment:
It's possible for the libraryURI ivar to not be null right?

Reply:
Addressed in xiao feedback 3 - added libraryuri support.